index.ts

```diff
@@ -271,10 +281,55 @@ app.post("/quiz/:type/question/:questionId", (req, res) => {
     res.redirect(`/quiz/${typeOfQuiz}/question/${questionId + 1}`);
 })
 
-app.get("/quiz/:type/score", (req, res) => {
-    // e.g. http://localhost:3000/quiz/10rounds/score
-    res.render("score");
-})
+app.get("/quiz/:type/score", async (req, res) => {
+    const typeOfQuiz = req.params.type;
+    let scores: number[] = [];
+    let highScore: number;
+    
+    if (user === null) {
+        throw "User not found";
+    } else {
+        switch (typeOfQuiz) {
+            case "tenrounds":
+                scores = questions.map(q => {
+                    if (q.correct_character === q.answer_character && q.correct_movie === q.answer_movie) {
+                        return 1;
+                    } else if (q.correct_character === q.answer_character || q.correct_movie === q.answer_movie) {
+                        return 0.5;
+                    } else {
+                        return 0;
+                    }
+                });
+                highScore = user.highscore_tenrounds;
+                break;
+            case "suddendeath":
+                let consecutiveCorrect = 0;
+                scores = questions.map(q => {
+                    if (q.correct_character === q.answer_character && q.correct_movie === q.answer_movie) {
+                        consecutiveCorrect++;
+                        return 1;
+                    } else {
+                        return 0;
+                    }
+                });
+                highScore = user.highscore_suddendeath;
+                break;
+            default:
+                break;
+        }
+    }
+    
+    let sumOfScores: number = scores.reduce((prev, curr) => prev + curr, 0);
+
+    //highScore used before being assigned error. Ik zit vast met hoe ik dit het best kan oplossen.
+    if (highScore < sumOfScores) {
+        await createNewHighScore(user, typeOfQuiz, sumOfScores);
+    }
+
+    await loadUser(user.username);
+    res.render("score", { sumOfScores, highScore });
+});
+
```

Comment:
This was really well done honestly, all the review comments I made were rather minor. Score functionality works perfectly as far as I can tell, and writing new high score to database also works.